src/components/Dropdown/Dropdown.tsx

```diff
@@ -300,17 +316,29 @@ const Dropdown = ({
     (e: React.FocusEvent) => {
       e.preventDefault();
       const target = e.nativeEvent.relatedTarget as HTMLElement | null;
+
       // check if we're focusing on something we don't control
-      if (!target || (target.id && !target.id.startsWith(name))) {
+      if (
+        !target ||
+        !target.getAttribute('data-test-id') ||
+        !(target.getAttribute('data-test-id') || '').startsWith(internalId.current)
```

Comment:
Does this work as intended if we focus on another foundry element that also has a data-test-id?

Reply:
Yes, because the internalId is either from a nanoid or defined by the user (and expected to be unique)

If the user passes two dropdowns the same dataTestId prop then it wouldn't work as intended.

Maybe it's worth not letting the user customize the focusing id string